### ISSUE TITLE
Indentation: custom annotation not checked

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#3134]
```
$ cat TestClass.java
     public class TestClass {}
     public interface TestInterface {}
     public enum TestEnum {}
     @interface MyAnnotation {} // line 4
     @interface MyAnnotation2 {
String author(); // line 6
     }

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<property name="tabWidth" value="4"/>

    <module name="Indentation">
      <property name="arrayInitIndent" value="4"/>
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="lineWrappingIndentation" value="4"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-6.17-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:1: 'class def modifier' have incorrect indentation level 5, expected level should be 0. [Indentation]
[ERROR] TestClass.java:2: 'interface def modifier' have incorrect indentation level 5, expected level should be 0. [Indentation]
[ERROR] TestClass.java:3: 'enum def modifier' have incorrect indentation level 5, expected level should be 0. [Indentation]
Audit done.
Checkstyle ends with 3 errors.
```

I was expecting a similar violation on line 4 as we require class, interface, and enum to all be at indent 0.
I was also expecting a violation at line 6 requiring a proper indentation, like expecting to be at column 9 (5 from brace + 4 for normal indentation).
##

**Comment by kukreja-vikramaditya** [comment]
I am on it

**Comment by kukreja-vikramaditya** [comment]
I resolved the issue by adding these two lines to the HandlerFactory
```java
        register(TokenTypes.ANNOTATION_DEF, ClassDefHandler.class);
        register(TokenTypes.ANNOTATION_FIELD_DEF, AnnotationDefHandler.class);
```

where newly created `AnnotationDefHandler` is based on `MethodDefHandler` (instead of `ClassDefHandler`)

However `mvn clean verify` fails as `HandlerFactory` is not allowed to have more than **30** executable statements. Can this limit be changed? If not, should `MethodDefHandler` be edited somehow to include `AnnotationDefHandler` as both of them are similar? (We will still be left with 1 extra execute command)

My fix here https://github.com/checkstyle/checkstyle/compare/master...kukreja-vikramaditya:fix-issue-3134 (PR not yet created as local build fails)

**Comment by romani** [comment]
@kukreja-vikramaditya , thanks a lot for hard work on this issue!!!
